xarray/test/test_variable.py

```diff
@@ -304,22 +309,23 @@ def test_equals_all_dtypes(self):
     def test_eq_all_dtypes(self):
         # ensure that we don't choke on comparisons for which numpy returns
         # scalars
-        expected = self.cls('x', 3 * [False])
+        expected = self.cls('x', 3 * [False]).to_base_variable()
         for v, _ in self.example_1d_objects():
             actual = 'z' == v
             self.assertVariableIdentical(expected, actual)
             actual = ~('z' != v)
             self.assertVariableIdentical(expected, actual)
 
     def test_encoding_preserved(self):
-        expected = self.cls('x', range(3), {'foo': 1}, {'bar': 2})
+        expected = self.cls('x', range(3), {'foo': 1}, {'bar': 2}).to_base_variable()
         for actual in [expected.T,
                        expected[...],
                        expected.squeeze(),
                        expected.isel(x=slice(None)),
                        expected.expand_dims({'x': 3}),
                        expected.copy(deep=True),
                        expected.copy(deep=False)]:
+
             self.assertVariableIdentical(expected, actual)
```

Comment:
Use `.to_base_variable()` in the assert here (on both sides). Otherwise this test doesn't have any coverage for `IndexVariable` objects.